packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -889,6 +895,60 @@ Future<void> _writeWebPluginRegistrant(FlutterProject project, List<Plugin> plug
   }
 }
 
+/// For each platform that uses them, creates symlinks within the platform
+/// directory to each plugin used on that platform.
+///
+/// This uses [project.flutterPluginsDependenciesFile], so should only be run
+/// after refreshPluginList has been run since the last plugin change.
+void createPluginSymlinks(FlutterProject project) {
+  Map<String, dynamic> platformPlugins;
+  final String pluginFileContent = _readFileContent(project.flutterPluginsDependenciesFile);
+  if (pluginFileContent != null) {
+    final Map<String, dynamic> pluginInfo = json.decode(pluginFileContent) as Map<String, dynamic>;
+    platformPlugins = pluginInfo[_kFlutterPluginsPluginListKey] as Map<String, dynamic>;
+  }
+  platformPlugins ??= <String, dynamic>{};
+
+  if (featureFlags.isWindowsEnabled && project.windows.existsSync()) {
+    _createPlatformPluginSymlinks(
+      project.windows.pluginSymlinkDirectory,
+      platformPlugins[project.windows.pluginConfigKey] as List<dynamic>);
+  }
+  if (featureFlags.isLinuxEnabled && project.linux.existsSync()) {
+    _createPlatformPluginSymlinks(
+      project.linux.pluginSymlinkDirectory,
+      platformPlugins[project.linux.pluginConfigKey] as List<dynamic>);
+  }
+}
+
+/// Replaces [symlinkDirectory] with a directory containing symlinks to each plugin
+/// listed in [platformPlugins].
+void _createPlatformPluginSymlinks(Directory symlinkDirectory, List<dynamic> platformPlugins) {
+  // Start fresh each time to avoid stale links.
```

Comment:
Is there some data we could track to avoid blowing the whole thing away? It seems like this might come back to bite us if we induce dependent build systems to do more work than necessary because of it.

Reply:
Changed to only fully recreate when the plugin list actually changes, and to do repairs (in case someone nukes `ephemeral/`) on each build. In practice I don't think it would affect build systems since it's a containing directory, not the project or sources, that is the symlink, but this version is still pretty low risk in terms of things getting out of sync.